### ISSUE TITLE
Unable to install Zeek-Exporter on Zeek 3.2.0

### ISSUE THREAD

**Issue by pisces-period** [esnet/zeek-exporter#2]
Hi,

I have a standalone Zeek Cluster which I'd like to enrich by using the zeek-exporter plugin.

However, the installation is failing and I'm not sure why.

I do have all the required packages that satisfy the pre-requisites (see the environment section below). 

Am I missing a step?

Thanks in advance!

**** **Environment Information** ****

__zeek version:__

`zeek version 3.2.0-dev.277`

__zeek package manager version:__

`zkg 2.1.0`

__zkg install output:__
```
$ zkg install zeek-exporter

The following packages will be INSTALLED:
  zeek/esnet/zeek-exporter (0.1.0)

Verify the following REQUIRED external dependencies:
(Ensure their installation on all relevant systems before proceeding):
  from zeek/esnet/zeek-exporter (0.1.0):
    cmake >=3.5
    libcurl-devel *

Proceed? [Y/n] Y
Running unit tests for "zeek/esnet/zeek-exporter"
error: failed to run tests for zeek/esnet/zeek-exporter: package build_command failed, see log in /home/ubuntu/.zkg/logs/zeek-exporter-build.log
Proceed to install anyway? [N/y] N
```

__zeek-exporter-build.log output:__
```
$ cat ~/.zkg/logs/zeek-exporter-build.log


=== STDERR ===
make: [build-it] Error 1 (ignored)
/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/src/Plugin.cc: In member function ‘void plugin::ESnet_Zeek_Exporter::Plugin::AddlArgumentPopulation(const char*, val_list*, std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> >&)’:
/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/src/Plugin.cc:82:62: error: invalid use of incomplete type ‘const class BroString’
         const char* arg_str = (*args)[arg_offset]->AsString()->CheckString();
                                                              ^~
In file included from /usr/local/zeek/include/zeek/Reporter.h:14:0,
                 from /usr/local/zeek/include/zeek/threading/Queue.h:10,
                 from /usr/local/zeek/include/zeek/threading/MsgThread.h:7,
                 from /usr/local/zeek/include/zeek/logging/WriterBackend.h:7,
                 from /usr/local/zeek/include/zeek/plugin/Plugin.h:10,
                 from /home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/src/Plugin.h:12,
                 from /home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/src/Plugin.cc:8:
/usr/local/zeek/include/zeek/IPAddr.h:14:7: note: forward declaration of ‘class BroString’
 class BroString;
       ^~~~~~~~~
/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/src/Plugin.cc:89:64: error: invalid use of incomplete type ‘const class BroString’
         const char* addl_str = (*args)[addl_offset]->AsString()->CheckString();
                                                                ^~
In file included from /usr/local/zeek/include/zeek/Reporter.h:14:0,
                 from /usr/local/zeek/include/zeek/threading/Queue.h:10,
                 from /usr/local/zeek/include/zeek/threading/MsgThread.h:7,
                 from /usr/local/zeek/include/zeek/logging/WriterBackend.h:7,
                 from /usr/local/zeek/include/zeek/plugin/Plugin.h:10,
                 from /home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/src/Plugin.h:12,
                 from /home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/src/Plugin.cc:8:
/usr/local/zeek/include/zeek/IPAddr.h:14:7: note: forward declaration of ‘class BroString’
 class BroString;
       ^~~~~~~~~
/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/src/Plugin.cc: In member function ‘virtual std::pair<bool, Val*> plugin::ESnet_Zeek_Exporter::Plugin::HookCallFunction(const Func*, Frame*, val_list*)’:
/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/src/Plugin.cc:131:41: error: cannot convert ‘IntrusivePtr<Val>’ to ‘Val*’ in initialization
     Val* result = func->Call(args, frame);
                                         ^
/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/src/Plugin.cc:147:38: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if ( child_func_durations.size() > ( my_func_depth + 1 ) )
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/src/Plugin.cc:196:66: error: invalid use of incomplete type ‘const class BroString’
             arg_events.insert({std::string((*args)[0]->AsString()->CheckString()), make_tuple(arg_val, addl_val)});
                                                                  ^~
In file included from /usr/local/zeek/include/zeek/Reporter.h:14:0,
                 from /usr/local/zeek/include/zeek/threading/Queue.h:10,
                 from /usr/local/zeek/include/zeek/threading/MsgThread.h:7,
                 from /usr/local/zeek/include/zeek/logging/WriterBackend.h:7,
                 from /usr/local/zeek/include/zeek/plugin/Plugin.h:10,
                 from /home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/src/Plugin.h:12,
                 from /home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/src/Plugin.cc:8:
/usr/local/zeek/include/zeek/IPAddr.h:14:7: note: forward declaration of ‘class BroString’
 class BroString;
       ^~~~~~~~~
/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/src/Plugin.cc:196:114: error: no matching function for call to ‘std::map<std::__cxx11::basic_string<char>, std::tuple<int, int> >::insert(<brace-enclosed initializer list>)’
             arg_events.insert({std::string((*args)[0]->AsString()->CheckString()), make_tuple(arg_val, addl_val)});
                                                                                                                  ^
In file included from /usr/include/c++/7/map:61:0,
                 from /home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/3rdparty/prometheus-cpp/core/include/prometheus/registry.h:3,
                 from /home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/3rdparty/prometheus-cpp/pull/include/prometheus/exposer.h:10,
                 from /home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/src/Plugin.cc:4:
/usr/include/c++/7/bits/stl_map.h:621:7: note: candidate: std::map<_Key, _Tp, _Compare, _Alloc>::insert_return_type std::map<_Key, _Tp, _Compare, _Alloc>::insert(std::map<_Key, _Tp, _Compare, _Alloc>::node_type&&) [with _Key = std::__cxx11::basic_string<char>; _Tp = std::tuple<int, int>; _Compare = std::less<std::__cxx11::basic_string<char> >; _Alloc = std::allocator<std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> > >; std::map<_Key, _Tp, _Compare, _Alloc>::insert_return_type = std::_Node_insert_return<std::_Rb_tree_iterator<std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> > >, std::_Node_handle<std::__cxx11::basic_string<char>, std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> >, std::allocator<std::_Rb_tree_node<std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> > > > > >; std::map<_Key, _Tp, _Compare, _Alloc>::node_type = std::_Node_handle<std::__cxx11::basic_string<char>, std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> >, std::allocator<std::_Rb_tree_node<std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> > > > >]
       insert(node_type&& __nh)
       ^~~~~~
/usr/include/c++/7/bits/stl_map.h:621:7: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘std::map<std::__cxx11::basic_string<char>, std::tuple<int, int> >::node_type&& {aka std::_Node_handle<std::__cxx11::basic_string<char>, std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> >, std::allocator<std::_Rb_tree_node<std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> > > > >&&}’
/usr/include/c++/7/bits/stl_map.h:626:7: note: candidate: std::map<_Key, _Tp, _Compare, _Alloc>::iterator std::map<_Key, _Tp, _Compare, _Alloc>::insert(std::map<_Key, _Tp, _Compare, _Alloc>::const_iterator, std::map<_Key, _Tp, _Compare, _Alloc>::node_type&&) [with _Key = std::__cxx11::basic_string<char>; _Tp = std::tuple<int, int>; _Compare = std::less<std::__cxx11::basic_string<char> >; _Alloc = std::allocator<std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> > >; std::map<_Key, _Tp, _Compare, _Alloc>::iterator = std::_Rb_tree_iterator<std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> > >; std::map<_Key, _Tp, _Compare, _Alloc>::const_iterator = std::_Rb_tree_const_iterator<std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> > >; std::map<_Key, _Tp, _Compare, _Alloc>::node_type = std::_Node_handle<std::__cxx11::basic_string<char>, std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> >, std::allocator<std::_Rb_tree_node<std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> > > > >]
       insert(const_iterator __hint, node_type&& __nh)
       ^~~~~~
/usr/include/c++/7/bits/stl_map.h:626:7: note:   candidate expects 2 arguments, 1 provided
/usr/include/c++/7/bits/stl_map.h:795:7: note: candidate: std::pair<typename std::_Rb_tree<_Key, std::pair<const _Key, _Tp>, std::_Select1st<std::pair<const _Key, _Tp> >, _Compare, typename __gnu_cxx::__alloc_traits<_Allocator>::rebind<std::pair<const _Key, _Tp> >::other>::iterator, bool> std::map<_Key, _Tp, _Compare, _Alloc>::insert(const value_type&) [with _Key = std::__cxx11::basic_string<char>; _Tp = std::tuple<int, int>; _Compare = std::less<std::__cxx11::basic_string<char> >; _Alloc = std::allocator<std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> > >; typename std::_Rb_tree<_Key, std::pair<const _Key, _Tp>, std::_Select1st<std::pair<const _Key, _Tp> >, _Compare, typename __gnu_cxx::__alloc_traits<_Allocator>::rebind<std::pair<const _Key, _Tp> >::other>::iterator = std::_Rb_tree_iterator<std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> > >; std::map<_Key, _Tp, _Compare, _Alloc>::value_type = std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> >]
       insert(const value_type& __x)
       ^~~~~~
/usr/include/c++/7/bits/stl_map.h:795:7: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const value_type& {aka const std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> >&}’
/usr/include/c++/7/bits/stl_map.h:802:7: note: candidate: std::pair<typename std::_Rb_tree<_Key, std::pair<const _Key, _Tp>, std::_Select1st<std::pair<const _Key, _Tp> >, _Compare, typename __gnu_cxx::__alloc_traits<_Allocator>::rebind<std::pair<const _Key, _Tp> >::other>::iterator, bool> std::map<_Key, _Tp, _Compare, _Alloc>::insert(std::map<_Key, _Tp, _Compare, _Alloc>::value_type&&) [with _Key = std::__cxx11::basic_string<char>; _Tp = std::tuple<int, int>; _Compare = std::less<std::__cxx11::basic_string<char> >; _Alloc = std::allocator<std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> > >; typename std::_Rb_tree<_Key, std::pair<const _Key, _Tp>, std::_Select1st<std::pair<const _Key, _Tp> >, _Compare, typename __gnu_cxx::__alloc_traits<_Allocator>::rebind<std::pair<const _Key, _Tp> >::other>::iterator = std::_Rb_tree_iterator<std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> > >; std::map<_Key, _Tp, _Compare, _Alloc>::value_type = std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> >]
       insert(value_type&& __x)
       ^~~~~~
/usr/include/c++/7/bits/stl_map.h:802:7: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘std::map<std::__cxx11::basic_string<char>, std::tuple<int, int> >::value_type&& {aka std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> >&&}’
/usr/include/c++/7/bits/stl_map.h:808:2: note: candidate: template<class _Pair> std::__enable_if_t<std::is_constructible<std::pair<const _Key, _Tp>, _Pair>::value, std::pair<typename std::_Rb_tree<_Key, std::pair<const _Key, _Tp>, std::_Select1st<std::pair<const _Key, _Tp> >, _Compare, typename __gnu_cxx::__alloc_traits<_Allocator>::rebind<std::pair<const _Key, _Tp> >::other>::iterator, bool> > std::map<_Key, _Tp, _Compare, _Alloc>::insert(_Pair&&) [with _Pair = _Pair; _Key = std::__cxx11::basic_string<char>; _Tp = std::tuple<int, int>; _Compare = std::less<std::__cxx11::basic_string<char> >; _Alloc = std::allocator<std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> > >]
  insert(_Pair&& __x)
  ^~~~~~
/usr/include/c++/7/bits/stl_map.h:808:2: note:   template argument deduction/substitution failed:
/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/src/Plugin.cc:196:114: note:   couldn't deduce template parameter ‘_Pair’
             arg_events.insert({std::string((*args)[0]->AsString()->CheckString()), make_tuple(arg_val, addl_val)});
                                                                                                                  ^
In file included from /usr/include/c++/7/map:61:0,
                 from /home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/3rdparty/prometheus-cpp/core/include/prometheus/registry.h:3,
                 from /home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/3rdparty/prometheus-cpp/pull/include/prometheus/exposer.h:10,
                 from /home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/src/Plugin.cc:4:
/usr/include/c++/7/bits/stl_map.h:822:7: note: candidate: void std::map<_Key, _Tp, _Compare, _Alloc>::insert(std::initializer_list<std::pair<const _Key, _Tp> >) [with _Key = std::__cxx11::basic_string<char>; _Tp = std::tuple<int, int>; _Compare = std::less<std::__cxx11::basic_string<char> >; _Alloc = std::allocator<std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> > >]
       insert(std::initializer_list<value_type> __list)
       ^~~~~~
/usr/include/c++/7/bits/stl_map.h:822:7: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘std::initializer_list<std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> > >’
/usr/include/c++/7/bits/stl_map.h:852:7: note: candidate: std::map<_Key, _Tp, _Compare, _Alloc>::iterator std::map<_Key, _Tp, _Compare, _Alloc>::insert(std::map<_Key, _Tp, _Compare, _Alloc>::const_iterator, const value_type&) [with _Key = std::__cxx11::basic_string<char>; _Tp = std::tuple<int, int>; _Compare = std::less<std::__cxx11::basic_string<char> >; _Alloc = std::allocator<std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> > >; std::map<_Key, _Tp, _Compare, _Alloc>::iterator = std::_Rb_tree_iterator<std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> > >; std::map<_Key, _Tp, _Compare, _Alloc>::const_iterator = std::_Rb_tree_const_iterator<std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> > >; std::map<_Key, _Tp, _Compare, _Alloc>::value_type = std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> >]
       insert(const_iterator __position, const value_type& __x)
       ^~~~~~
/usr/include/c++/7/bits/stl_map.h:852:7: note:   candidate expects 2 arguments, 1 provided
/usr/include/c++/7/bits/stl_map.h:862:7: note: candidate: std::map<_Key, _Tp, _Compare, _Alloc>::iterator std::map<_Key, _Tp, _Compare, _Alloc>::insert(std::map<_Key, _Tp, _Compare, _Alloc>::const_iterator, std::map<_Key, _Tp, _Compare, _Alloc>::value_type&&) [with _Key = std::__cxx11::basic_string<char>; _Tp = std::tuple<int, int>; _Compare = std::less<std::__cxx11::basic_string<char> >; _Alloc = std::allocator<std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> > >; std::map<_Key, _Tp, _Compare, _Alloc>::iterator = std::_Rb_tree_iterator<std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> > >; std::map<_Key, _Tp, _Compare, _Alloc>::const_iterator = std::_Rb_tree_const_iterator<std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> > >; std::map<_Key, _Tp, _Compare, _Alloc>::value_type = std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> >]
       insert(const_iterator __position, value_type&& __x)
       ^~~~~~
/usr/include/c++/7/bits/stl_map.h:862:7: note:   candidate expects 2 arguments, 1 provided
/usr/include/c++/7/bits/stl_map.h:867:2: note: candidate: template<class _Pair> std::__enable_if_t<std::is_constructible<std::pair<const _Key, _Tp>, _Pair>::value, typename std::_Rb_tree<_Key, std::pair<const _Key, _Tp>, std::_Select1st<std::pair<const _Key, _Tp> >, _Compare, typename __gnu_cxx::__alloc_traits<_Allocator>::rebind<std::pair<const _Key, _Tp> >::other>::iterator> std::map<_Key, _Tp, _Compare, _Alloc>::insert(std::map<_Key, _Tp, _Compare, _Alloc>::const_iterator, _Pair&&) [with _Pair = _Pair; _Key = std::__cxx11::basic_string<char>; _Tp = std::tuple<int, int>; _Compare = std::less<std::__cxx11::basic_string<char> >; _Alloc = std::allocator<std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> > >]
  insert(const_iterator __position, _Pair&& __x)
  ^~~~~~
/usr/include/c++/7/bits/stl_map.h:867:2: note:   template argument deduction/substitution failed:
/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/src/Plugin.cc:196:114: note:   candidate expects 2 arguments, 1 provided
             arg_events.insert({std::string((*args)[0]->AsString()->CheckString()), make_tuple(arg_val, addl_val)});
                                                                                                                  ^
In file included from /usr/include/c++/7/map:61:0,
                 from /home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/3rdparty/prometheus-cpp/core/include/prometheus/registry.h:3,
                 from /home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/3rdparty/prometheus-cpp/pull/include/prometheus/exposer.h:10,
                 from /home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/src/Plugin.cc:4:
/usr/include/c++/7/bits/stl_map.h:885:2: note: candidate: template<class _InputIterator> void std::map<_Key, _Tp, _Compare, _Alloc>::insert(_InputIterator, _InputIterator) [with _InputIterator = _InputIterator; _Key = std::__cxx11::basic_string<char>; _Tp = std::tuple<int, int>; _Compare = std::less<std::__cxx11::basic_string<char> >; _Alloc = std::allocator<std::pair<const std::__cxx11::basic_string<char>, std::tuple<int, int> > >]
  insert(_InputIterator __first, _InputIterator __last)
  ^~~~~~
/usr/include/c++/7/bits/stl_map.h:885:2: note:   template argument deduction/substitution failed:
/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/src/Plugin.cc:196:114: note:   candidate expects 2 arguments, 1 provided
             arg_events.insert({std::string((*args)[0]->AsString()->CheckString()), make_tuple(arg_val, addl_val)});
                                                                                                                  ^
make[3]: *** [CMakeFiles/ESnet-Zeek_Exporter.linux-x86_64.dir/src/Plugin.cc.o] Error 1
make[2]: *** [CMakeFiles/ESnet-Zeek_Exporter.linux-x86_64.dir/all] Error 2
make[1]: *** [all] Error 2
make: *** [build-it] Error 2
=== STDOUT ===
Build Directory        : build
Zeek Source Directory   : 
-- The C compiler identification is GNU 7.5.0
-- The CXX compiler identification is GNU 7.5.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test cxx17_already_works
-- Performing Test cxx17_already_works - Failed
-- Performing Test cxx17_works
-- Performing Test cxx17_works - Success
-- Found BinPAC: /usr/local/zeek/bin/binpac  
-- Found CAF: 0.17.4  found components:  core io openssl 
-- Found Broker: /usr/local/zeek/lib/libbroker.so  
-- Zeek executable      : /usr/local/zeek/bin/zeek
-- Zeek source          : 
-- Zeek build           : 
-- Zeek install prefix  : /usr/local/zeek
-- Zeek plugin directory: /usr/local/zeek/lib/zeek/plugins
-- Zeek debug mode      : 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Could NOT find benchmark (missing: benchmark_DIR)
-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found version "1.2.11") 
-- Found CURL: /usr/lib/x86_64-linux-gnu/libcurl.so (found version "7.58.0") 
-- The following features have been enabled:

 * Pull, support for pulling metrics
 * Push, support for pushing metrics to a push-gateway
 * Compression, support for zlib compression of metrics

-- The following REQUIRED packages have been found:

 * BinPAC
 * CAF
 * Broker
 * googlemock-3rdparty
 * civetweb-3rdparty
 * ZLIB
 * CURL

-- The following OPTIONAL packages have not been found:

 * benchmark

-- Configuring done
-- Generating done
-- Build files have been written to: /home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build
Makefile:11: recipe for target 'build-it' failed
( cd build && make )
make[1]: Entering directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
make[2]: Entering directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
make[3]: Entering directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
Scanning dependencies of target civetweb
make[3]: Leaving directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
make[3]: Entering directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
[  2%] Building CXX object 3rdparty/prometheus-cpp/pull/CMakeFiles/civetweb.dir/__/3rdparty/civetweb/src/CivetServer.cpp.o
[  4%] Building C object 3rdparty/prometheus-cpp/pull/CMakeFiles/civetweb.dir/__/3rdparty/civetweb/src/civetweb.c.o
make[3]: Leaving directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
[  4%] Built target civetweb
make[3]: Entering directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
Scanning dependencies of target core
make[3]: Leaving directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
make[3]: Entering directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
[  6%] Building CXX object 3rdparty/prometheus-cpp/core/CMakeFiles/core.dir/src/check_names.cc.o
[  8%] Building CXX object 3rdparty/prometheus-cpp/core/CMakeFiles/core.dir/src/counter.cc.o
[ 10%] Building CXX object 3rdparty/prometheus-cpp/core/CMakeFiles/core.dir/src/detail/ckms_quantiles.cc.o
[ 12%] Building CXX object 3rdparty/prometheus-cpp/core/CMakeFiles/core.dir/src/detail/time_window_quantiles.cc.o
[ 14%] Building CXX object 3rdparty/prometheus-cpp/core/CMakeFiles/core.dir/src/detail/utils.cc.o
[ 17%] Building CXX object 3rdparty/prometheus-cpp/core/CMakeFiles/core.dir/src/gauge.cc.o
[ 19%] Building CXX object 3rdparty/prometheus-cpp/core/CMakeFiles/core.dir/src/histogram.cc.o
[ 21%] Building CXX object 3rdparty/prometheus-cpp/core/CMakeFiles/core.dir/src/registry.cc.o
[ 23%] Building CXX object 3rdparty/prometheus-cpp/core/CMakeFiles/core.dir/src/serializer.cc.o
[ 25%] Building CXX object 3rdparty/prometheus-cpp/core/CMakeFiles/core.dir/src/summary.cc.o
[ 27%] Building CXX object 3rdparty/prometheus-cpp/core/CMakeFiles/core.dir/src/text_serializer.cc.o
[ 29%] Linking CXX static library libprometheus-cpp-core.a
make[3]: Leaving directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
[ 29%] Built target core
make[3]: Entering directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
Scanning dependencies of target pull
make[3]: Leaving directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
make[3]: Entering directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
[ 31%] Building CXX object 3rdparty/prometheus-cpp/pull/CMakeFiles/pull.dir/src/exposer.cc.o
[ 34%] Building CXX object 3rdparty/prometheus-cpp/pull/CMakeFiles/pull.dir/src/handler.cc.o
[ 36%] Linking CXX static library libprometheus-cpp-pull.a
make[3]: Leaving directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
[ 36%] Built target pull
make[3]: Entering directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
Scanning dependencies of target bif-plugin-ESnet_Zeek_Exporter-zeek_exporter.bif
make[3]: Leaving directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
make[3]: Entering directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
[ 38%] [BIFCL] Processing src/zeek_exporter.bif
make[3]: Leaving directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
[ 38%] Built target bif-plugin-ESnet_Zeek_Exporter-zeek_exporter.bif
make[3]: Entering directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
Scanning dependencies of target generate_outputs
make[3]: Leaving directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
[ 38%] Built target generate_outputs
make[3]: Entering directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
Scanning dependencies of target bif-init-ESnet_Zeek_Exporter
make[3]: Leaving directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
make[3]: Entering directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
make[3]: Leaving directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
[ 38%] Built target bif-init-ESnet_Zeek_Exporter
make[3]: Entering directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
Scanning dependencies of target copy-scripts-ESnet_Zeek_Exporter
make[3]: Leaving directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
make[3]: Entering directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
make[3]: Leaving directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
[ 38%] Built target copy-scripts-ESnet_Zeek_Exporter
make[3]: Entering directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
Scanning dependencies of target bro-plugin-ESnet_Zeek_Exporter
make[3]: Leaving directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
make[3]: Entering directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
[ 40%] Creating __bro_plugin__ for ESnet::Zeek_Exporter
make[3]: Leaving directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
[ 40%] Built target bro-plugin-ESnet_Zeek_Exporter
make[3]: Entering directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
Scanning dependencies of target ESnet-Zeek_Exporter.linux-x86_64
make[3]: Leaving directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
make[3]: Entering directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
[ 42%] Building CXX object CMakeFiles/ESnet-Zeek_Exporter.linux-x86_64.dir/src/Plugin.cc.o
CMakeFiles/ESnet-Zeek_Exporter.linux-x86_64.dir/build.make:81: recipe for target 'CMakeFiles/ESnet-Zeek_Exporter.linux-x86_64.dir/src/Plugin.cc.o' failed
make[3]: Leaving directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
CMakeFiles/Makefile2:239: recipe for target 'CMakeFiles/ESnet-Zeek_Exporter.linux-x86_64.dir/all' failed
make[2]: Leaving directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
Makefile:151: recipe for target 'all' failed
make[1]: Leaving directory '/home/ubuntu/.zkg/testing/zeek-exporter/clones/zeek-exporter/build'
Makefile:11: recipe for target 'build-it' failed
```

**Comment by grigorescu** [comment]
Can you try the latest master? I just merged a branch with some 3.2 fixes.

**Comment by grigorescu** [comment]
Fixed in 5eb71365e37f4307ed8f52cfebcbad7cc388bdc1.

Also, I added some automated tests to better catch such issues.
